### PR TITLE
新增映射文档

### PR DESCRIPTION
**Pull request by Xuxuanang**
https://github.com/PaddlePaddle/PaConvert/pull/583

### CODE REVIEW

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch/torch._foreach_round_.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# PyTorch 写法
+torch._foreach_round_(tensors)
+
+# Paddle 写法
+[paddle.assign(paddle.round(x), x) for x in tensors]
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/torch/torch._foreach_round_.md]: `[x.round_() for x in tensors]` 更简洁

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.all_gather_into_tensor.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+## [ 输入参数用法不一致 ]torch.distributed.all_gather_into_tensor
+
+### [torch.distributed.all_gather_into_tensor](https://docs.pytorch.org/docs/stable/distributed.html#torch.distributed.all_gather_into_tensor)
+
+```python
+torch.distributed.all_gather_into_tensor(output_tensor, input_tensor, group=None, async_op=False)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.all_gather_into_tensor.md:6]: torch的output_tensor是一个Tensor，paddle的是一个Tensor list，有没有这个区别？可能需要实际写代码尝试下

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.all_gather_into_tensor.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+```
+
+### [paddle.distributed.stream.all_gather](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/distributed/stream/all_gather_cn.html#all-gather)
+
+```python
+paddle.distributed.stream.all_gather(tensor_or_tensor_list, tensor, group=None, sync_op=True, use_calc_stream=False)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.all_gather_into_tensor.md]: 这个对应到`paddle.distributed.all_gather`吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+#### input 参数：输入的张量
+
+```python
+# PyTorch 写法:
+torch.distributed.reduce_scatter_tensor(data1, input)
+
+# Paddle 写法:
+input_list = [input[i] for i in range(input.shape[0])]
+paddle.distributed.reduce_scatter(data1, input_list)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md]: 也不用这样转写吧，tensor_list里可以只有一个tesnor，比如input_list=[input]

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch/torch._foreach_round_.md (hunk 1 above)
Comment (Xuxuanang) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch/torch._foreach_round_.md]: 那这个需要再新写一个Matcher吗，还是只修改映射文档，Matcher复用原来的

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md (hunk 4 above)
Comment (Xuxuanang) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md]: 我个人感觉应该不行吧 PyTorch输入的是没有切分好的 会自动切分后分发到各进程 但是PaddlePaddle需要手动切分一下

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch/torch._foreach_round_.md (hunk 1 above)
Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch/torch._foreach_round_.md]: > 那这个需要再新写一个Matcher吗，还是只修改映射文档，Matcher复用原来的

新写一个Matcher吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+### 转写示例
+
+#### input 参数：输入的张量
+
+```python
+# PyTorch 写法:
+torch.distributed.reduce_scatter_tensor(data1, input)
+
+# Paddle 写法:
+input_list = [input[i] for i in range(input.shape[0])]
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md]: 这个应该是按`world_size` 来切分的，并且第0维如果是等于world_size，是按unstack来的，如果不等于world_size，是按split来的。这些细节要体现出来。

infoflow 2025-05-26 11-34-07

File: docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md [hunk 6]
```diff
@@ -1055,19 +1055,18 @@
...
+| NOT-IMPLEMENTED-ITEM(`torch.distributed.get_process_group_ranks`, https://docs.pytorch.org/docs/stable/distributed.html#torch.distributed.get_process_group_ranks, 可新增，且框架底层有相关设计，成本低) |
+| NOT-IMPLEMENTED-ITEM(`torch.nn.utils.fuse_conv_bn_eval`, https://pytorch.org/docs/stable/generated/torch.nn.utils.fuse_conv_bn_eval.html#torch-nn-utils-fuse-conv-bn-eval, 可新增，且框架底层有相关设计，成本低) |
+| NOT-IMPLEMENTED-ITEM(`torch.nn.utils.fuse_conv_bn_weights`, https://pytorch.org/docs/stable/generated/torch.nn.utils.fuse_conv_bn_weights.html#torch-nn-utils-fuse-conv-bn-weights, 可新增，且框架底层有相关设计，成本低) |
+| NOT-IMPLEMENTED-ITEM(`torch.nn.utils.fuse_linear_bn_eval`, https://pytorch.org/docs/stable/generated/torch.nn.utils.fuse_linear_bn_eval.html#torch-nn-utils-fuse-linear-bn-eval, 可新增，且框架底层有相关设计，成本低) |
+| NOT-IMPLEMENTED-ITEM(`torch.nn.utils.fuse_linear_bn_weights`, https://pytorch.org/docs/stable/generated/torch.nn.utils.fuse_linear_bn_weights.html#torch-nn-utils-fuse-linear-bn-weights, 可新增，且框架底层有相关设计，成本低) |
+| NOT-IMPLEMENTED-ITEM(`torch.cuda.is_initialized`, https://pytorch.org/docs/stable/generated/torch.cuda.is_initialized.html#torch-cuda-is-initialized, 可新增，但框架底层无相关设计，成本高) |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md]: 这个应该好搞吧，随便找一个返回cuda状态之类的API，就也能算作 **cuda是否初始化的判断**

Comment (Xuxuanang) [reply docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md]: 能直接映射到paddle.is_compiled_with_cuda()上面吗，感觉PaddlePaddle没有这种完全等价的

Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/pytorch_api_mapping_cn.md]: > 能直接映射到paddle.is_compiled_with_cuda()上面吗，感觉PaddlePaddle没有这种完全等价的

用这个也行

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.all_gather_into_tensor.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+# Paddle 写法
+import paddle.distributed as dist
+tensor_list = []
+dist.all_gather(tensor_list=tensor_list, tensor=input_tensor)
+if paddle.distributed.get_world_size() * input_tensor.shape[0] == output_tensor.shape[0]:
+    output_tensor = paddle.concat(tensor_list, axis=0)
+else:
+    output_tensor = paddle.stack(tensor_list, axis=0)
+```
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.all_gather_into_tensor.md:42]: 下个PR补一下async_op的转写

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+### 参数映射
+
+| PyTorch    | PaddlePaddle | 备注                                                                   |
+| ---------- | ------------ | ---------------------------------------------------------------------- |
+| output     | tensor       | 用于接收数据的 tensor，仅参数名不一致。                                |
+| input | tensor_list  | 将被规约和分发的 tensor 列表，PyTorh 为 tensor， Paddle 为 list， 需要转写。                         |
+| op         | op           | 归约的操作类型。                                                       |
+| group      | group        | 执行该操作的进程组实例。                                               |
+| async_op   | sync_op      | 该操作是否为异步或同步操作，PyTorch 和 Paddle 取值相反，需要转写。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md:25]: 转写方式取反即可

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+torch.distributed.reduce_scatter_tensor(output, input)
+
+# Paddle 写法:
+world_size = paddle.distributed.get_world_size()
+if input.shape[0] == world_size:
+    input_list = paddle.unstack(input, axis=0)
+else:
+    input_list = paddle.split(input, num_or_sections=world_size, axis=0)
+paddle.distributed.reduce_scatter(output, input_list)
+```
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.reduce_scatter_tensor.md:42]: 下个PR补一下async_op的转写

**Review by zhwesky2010 — APPROVED** [review]

**Comment by zhwesky2010** [comment]
下个PR改一下
